bot/exts/moderation/stream.py

```diff
@@ -0,0 +1,89 @@
+import datetime
+
+import discord
+from discord.ext import commands
+
+from bot.bot import Bot
+from bot.constants import Emojis, Roles, STAFF_ROLES, VideoPermission
+from bot.converters import Expiry
+from bot.utils.scheduling import Scheduler
+from bot.utils.time import format_infraction_with_duration
+
+# Constant error messages
+TIME_FORMAT_NOT_VALID = f"{Emojis.cross_mark}Please specify a valid time format ex. 10h or 1day."
+TIME_LESS_EQ_0 = f"{Emojis.cross_mark}Duration can not be a 0 or lower."
+USER_ALREADY_ALLOWED_TO_STREAM = f"{Emojis.cross_mark}This user can already stream."
+USER_ALREADY_NOT_ALLOWED_TO_STREAM = f"{Emojis.cross_mark}This user already can't stream."
+
+
+class Stream(commands.Cog):
+    """Grant and revoke streaming permissions from users."""
+
+    def __init__(self, bot: Bot):
+        self.bot = bot
+        self.scheduler = Scheduler(self.__class__.__name__)
+
+    @staticmethod
+    async def _remove_streaming_permission(schedule_user: discord.Member) -> None:
+        """Remove streaming permission from Member."""
+        await schedule_user.remove_roles(discord.Object(Roles.video), reason="Temporary streaming access revoked")
+
+    @commands.command(aliases=("streaming",))
+    @commands.has_any_role(*STAFF_ROLES)
+    async def stream(
+            self,
+            ctx: commands.Context,
+            user: discord.Member,
+            duration: Expiry =
+            datetime.datetime.utcnow() + datetime.timedelta(minutes=VideoPermission.default_permission_duration),
```

Comment:
Default values are only computed once. That happens when the function is defined. Therefore, `now()` will not actually be the time at which the command is invoked, but the time at which the function was defined. The only practical solution is to calculate this within the function rather than as a default value.

Reply:
Now it default to None and calculates time if duration is None